rust_src/src/editfns.rs

```diff
@@ -0,0 +1,14 @@
+//! Lisp functions pertaining to editing.
+
+use remacs_macros::lisp_fn;
+use lisp::{LispObject};
+use remacs_sys::{Fcurrent_buffer, EmacsInt};
+
+
+/// Return value of point, as an integer.
+#[lisp_fn]
+pub fn point() -> LispObject {
+    let buffer = LispObject::from_raw(unsafe { Fcurrent_buffer() });
+    let b = buffer.as_vectorlike().unwrap().as_buffer().unwrap();
```

Comment:
I think this can be just `let b = buffer.as_buffer().unwrap();`, since as_buffer() will do the as_vectorlike call for you.